### PR TITLE
Makes RND servers perform worse if they're running on superheated air, not better. Servers are supposed to be cooled, not burned.

### PR DESCRIPTION
**Pull request by silicons**
Also makes them not initialize twice.

### CODE REVIEW

**Comment by Armhulen** [comment]
It's another fix shoehorned in with balance change episode

File: code/modules/research/server.dm [hunk 1]
```diff
@@ -59,8 +57,9 @@
 			heat_health = min(100, heat_health + 1)
 		if(T0C to (T20C + 20))
 			heat_health = Clamp(heat_health, 0, 100)
-		if((T20C + 20) to (T0C + 70))
-			heat_health = max(0, heat_health - 1)
+		if((T20C + 20) to INFINITY)
+			var/excess = environment.temperature - (T20C + 20)
+			heat_health = max(0, -(excess/40))
 	if(heat_health <= 0)
 		/*griefProtection() This seems to get called twice before running any code that deletes/damages the server or it's files anwyay.
```
Comment (lordpidey) [inline code/modules/research/server.dm]: This should go back in, as this if loop has a chance to remove tech levels.

You might be able to put it in the `if(prob(1))` section, to reduce the number of times it's called.

something along the lines of
```
if(!updateRD)
     griefProtection()
updateRD++
T.level--
```

File: code/modules/research/server.dm [hunk 2]
```diff
@@ -59,8 +57,9 @@
 			heat_health = min(100, heat_health + 1)
 		if(T0C to (T20C + 20))
 			heat_health = Clamp(heat_health, 0, 100)
-		if((T20C + 20) to (T0C + 70))
-			heat_health = max(0, heat_health - 1)
+		if((T20C + 20) to INFINITY)
+			var/excess = environment.temperature - (T20C + 20)
+			heat_health = max(0, -(excess/40))
```
Comment (lordpidey) [inline code/modules/research/server.dm]: This max will always return 0, thus is pointless.


But even beyond that, the whole switch statement here is confusing and irrelevant.
The amount of overheating doesn't change how much the server loses, it's either overheating or it isn't.

Consider the following:

```
 	switch(environment.temperature)
 		if(0 to T0C)
  			overheating = FALSE
 		if((T20C + 20) to INFINITY)
 			overheating = TRUE
	if(overheating)
		//bad_stuff
```

With the way it currently is written, that is identical, and is MUCH easier to read.

Or an alternative method:
```
 	switch(environment.temperature)
 		if(0 to T0C)
  			heat_health++
 		if((T20C + 20) to INFINITY)
			var/excess = environment.temperature - (T20C + 20)
 			heat_health -= excess/40
	heat_health = Clamp(heat_health, 0, 100)
	if(heat_health <= 0)
		//bad_stuff
```

With this second method, the server will have a short grace period when overheating before it starts losing R&D levels.

**Review by lordpidey — CHANGES_REQUESTED** [review]

**Comment by kilkun** [comment]
this isn't really a balance change, this was always the intention.

**Comment by kevinz000** [comment]
rnd techweb pr covers this

**Pull request closed without merging**